vendor/fbtransform/transforms/reactDisplayName.js

```diff
@@ -72,11 +73,40 @@ function visitReactDisplayName(traverse, object, path, state) {
     right = object.init;
   }
 
+  var displayName;
   if (left && left.type === Syntax.MemberExpression) {
-    left = left.property;
+    if (left.object.type === Syntax.Identifier &&
+        left.object.name === 'exports') {
+      // Ignore the "exports" property access.
+      if (left.property.type === Syntax.Identifier) {
+        displayName = left.property.name;
+      }
+    } else {
+      var memberExpression = left;
+      var displayNamePath = [];
+      while (memberExpression) {
+        if (memberExpression.property.type === Syntax.Identifier) {
```

Comment:
(You need to check `memberExpression.computed` here or else you'll catch `foo[bar]`.)